nixos/modules/services/misc/transfer-sh.nix

```diff
@@ -0,0 +1,422 @@
+      enable = mkEnableOption "transfer-sh setup";
+
+      package = mkPackageOption pkgs "transfer-sh" { };
+
+      environmentFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = lib.mdDoc "Environment file as defined in {manpage}`systemd.exec(5)`.";
+      };
+
+      user = mkOption {
+        description = "user to run as";
+        default = "transfersh";
+        type = types.str;
+      };
+
+      group = mkOption {
+        description = "group to run as";
+        default = "transfersh";
+        type = types.str;
+      };
+
+      provider = mkOption {
+        description = "which storage provider to use (s3, storj, gdrive or local)";
+        default = "local";
+        type = types.enum [ "s3" "storj" "gdrive" "local" ];
+      };
+
+      address = mkOption {
+        description = "address to listen on";
+        default = "127.0.0.1";
+        type = types.str;
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = "Open the firewall port";
+      };
+
+      LISTENER = mkOption {
+        description = "port to use for http";
+        default = 6080;
+        type = types.port;
+      };
+
+      PROFILE_LISTENER = mkOption {
+        description = "port to use for profiler";
+        default = 6060;
+        type = types.nullOr types.port;
+      };
+
+      FORCE_HTTPS = mkOption {
+        description = "redirect to https";
+        default = false;
+        type = types.nullOr types.bool;
+      };
+
+      TLS_LISTENER = mkOption {
+        description = "port to use for https";
+        default = null;
+        type = types.nullOr types.port;
+      };
+
+      TLS_LISTENER_ONLY = mkOption {
+        description = "flag to enable tls listener only";
+        default = false;
+        type = types.nullOr types.bool;
+      };
+
+      TLS_CERT_FILE = mkOption {
+        description = "path to tls certificate";
+        default = null;
+        type = types.nullOr types.path;
+      };
+
+      TLS_PRIVATE_KEY = mkOption {
+        description = "path to tls private key";
+        default = null;
+        type = types.nullOr types.path;
+      };
+
+      HTTP_AUTH_USER = mkOption {
+        description = "user for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_PASS = mkOption {
+        description = "pass for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.str;
+      };
+
+      HTTP_AUTH_HTPASSWD = mkOption {
+        description = "htpasswd file path for basic http auth on upload";
+        default = null;
+        type = types.nullOr types.path;
+      };
+
+      HTTP_AUTH_IP_WHITELIST = mkOption {
+        description = "comma separated list of ips allowed to upload without being challenged an http auth";
+        default = [ ];
+        type = with types; listOf str;
```

Comment:
I'd prefer uses of `with` like this are also removed but its much less important than the top most one and is a preference thing.